src/gvmd.c

```diff
@@ -1184,30 +1184,33 @@ update_nvt_cache_retry ()
               }
             case SCANNER_TYPE_OPENVASD:
               {
-#if OPENVASD
-                int ret;
+                if (feature_enabled (FEATURE_ID_OPENVASD_SCANNER))
+                  {
+                    int ret;
+
+                    setproctitle ("openvasd: Updating NVT cache");
+                    ret = manage_update_nvt_cache_openvasd ();
+                    if (ret == 1)
+                      {
+                        g_message (
+                          "Rebuilding all NVTs because of a hash value mismatch");
+                        ret = update_or_rebuild_nvts (0);
+                        if (ret)
+                          g_warning ("%s: rebuild failed", __func__);
+                        else
+                          g_message ("%s: rebuild successful", __func__);
+                      }
 
-                setproctitle ("openvasd: Updating NVT cache");
-                ret = manage_update_nvt_cache_openvasd ();
-                if (ret == 1)
+                    gvm_close_sentry ();
+                    exit (ret);
+                  }
+                else
                   {
-                    g_message (
-                      "Rebuilding all NVTs because of a hash value mismatch");
-                    ret = update_or_rebuild_nvts (0);
-                    if (ret)
-                      g_warning ("%s: rebuild failed", __func__);
-                    else
-                      g_message ("%s: rebuild successful", __func__);
+                    g_critical ("%s: Default scanner is an openvasd one,"
+                                " but gvmd is not built to support this.",
```

Comment:
I think this should just say it's not enabled as openvasd may be disabled with the runtime flag.